Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py

```diff
@@ -697,18 +697,24 @@ def create_pull_request(cls, repository, args, branch_point, callback=None, unbl
             if cls.is_revert_commit(commits[0]) and update_issue:
                 cls.add_comment_to_reverted_commit_bug_tracker(repository, args, pr, commits[0])
 
-        if issue and update_issue:
-            log.info('Checking issue assignee...')
-            if issue.assignee != issue.tracker.me():
-                issue.assign(issue.tracker.me())
-                print('Assigning associated issue to {}'.format(issue.tracker.me()))
-            log.info('Checking for pull request link in associated issue...')
-            if pr.url and not any([pr.url in comment.content for comment in issue.comments]):
-                if issue.opened:
-                    issue.add_comment('Pull request: {}'.format(pr.url))
-                else:
-                    issue.open(why='Re-opening for pull request {}'.format(pr.url))
-                print('Posted pull request link to {}'.format(issue.link))
+        def comment_on_issue(issue):
```

Comment:
I feel like this may as well be a method on the class in its own right, because that only means additionally taking `pr` as an argument, and then there's slightly less complexity in what's currently a 333 line function?